golang/cosmos/x/vstorage/keeper/keeper.go

```diff
@@ -153,16 +161,14 @@ func (k Keeper) MigrateNoDataPlaceholders(ctx sdk.Context) {
 	// Copy empty keys first since cosmos stores do not support writing keys
 	// while an iterator is open over the domain
 	emptyKeys := []*([]byte){}
-	{
-		defer iterator.Close()
-		for ; iterator.Valid(); iterator.Next() {
-			rawValue := iterator.Value()
-			if bytes.Equal(rawValue, types.EncodedDataPrefix) {
-				key := iterator.Key()
-				emptyKeys = append(emptyKeys, &key)
-			}
+	for ; iterator.Valid(); iterator.Next() {
+		rawValue := iterator.Value()
+		if bytes.Equal(rawValue, types.EncodedDataPrefix) {
+			key := iterator.Key()
+			emptyKeys = append(emptyKeys, &key)
```

Comment:
I just noticed this use of pointers... are we **certain** that the iterator will never do anything with the memory referenced by `Key()`? It might be safer to define `emptyKeys := [][]byte{}` and update use accordingly.
```suggestion
			key := iterator.Key()
			emptyKeys = append(emptyKeys, bytes.Clone(key))
```

Reply:
`bytes.Clone` doesn't exists in go < 1.20, so had to manually do the copy.